CurrencyConverterControl/CurrencyConverterControl.cs

```diff
@@ -0,0 +1,251 @@
+﻿using CurrencyConverterControl.Data;
+using CurrencyConverterControl.Model;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace CurrencyConverterControl
+{
+    /// <summary>
+    /// Represents a custom WPF control for currency conversion.
+    /// This control extends the base Control class and provides functionality
+    /// for converting values between different currencies, along with a customizable
+    /// user interface.
+    /// </summary>
+    [TemplatePart(Name = "CmbDestination", Type = typeof(ComboBox))]
+    [TemplatePart(Name = "CmbSource", Type= typeof(ComboBox))]   
+    public class CurrencyConverterControl : Control, INotifyPropertyChanged
+    {
+        private static ICurrencyDataProvider CurrencyDataProvider;
+        private ComboBox _currencyDestination;
+        private ComboBox _currencySource;
+        
+        static CurrencyConverterControl()
+        {
+            DefaultStyleKeyProperty.OverrideMetadata(typeof(CurrencyConverterControl), new FrameworkPropertyMetadata(typeof(CurrencyConverterControl)));
+            CurrencyDataProvider = new CurrencyDataProvider();
+        }
+
+        #region Dependency Properties
+        /// <summary>
+        /// Gets or sets the input value for currency conversion within the CurrencyConverterControl.
+        /// </summary>
+        /// <remarks>
+        /// The InputValue property represents the value to be converted between different currencies.
+        /// It is bound to the user interface elements where users can input the amount for conversion.
+        /// </remarks>
+        [Description("Value to be converted")]
+        public double InputValue
+        {
+            get { return (double)GetValue(InputValueProperty); }
+            set { SetValue(InputValueProperty, value); }
+        }
+        public static readonly DependencyProperty InputValueProperty =
+            DependencyProperty.Register("InputValue", typeof(double), typeof(CurrencyConverterControl));
```

Comment:
please look at overloads and examples of usin DependencyProperty.Register method. The missing part here is callback for property changes. You need to define such callback in control and register it aloong with dependency property. Then, in callback method, you will be able to properly handle property changes regardless of the way they are changed. Callback will work in all cases:

- when end-user types something
- when application author sets this property from code
- when this property is bound to some object model in application's xaml